pennylane/devices/qubit/simulate.py

```diff
@@ -237,7 +259,303 @@ def simulate(
+
+
+def init_auxiliary_circuit(circuit: qml.tape.QuantumScript):
+    """Creates an auxiliary circuit to perform one-shot mid-circuit measurement calculations.
+
+    Measurements with non-trivial measurement values are removed from the script. VarianceMP
+    measurements are also replaced by SampleMP measurements, which are necessary to evaluate
+    the variance.
+
+    Args:
+        circuit (QuantumTape): The original QuantumScript
+
+    Returns:
+        QuantumScript: A copy of the circuit with modified measurements
+    """
+    new_measurements = []
+    for m in circuit.measurements:
+        if not m.mv:
+            if isinstance(m, VarianceMP):
+                new_measurements.append(SampleMP(obs=m.obs))
+            else:
+                new_measurements.append(m)
+    return qml.tape.QuantumScript(
+        circuit.operations, new_measurements, shots=1, trainable_params=circuit.trainable_params
+    )
+
+
+def simulate_one_shot_native_mcm(
+    circuit: qml.tape.QuantumScript,
+    rng=None,
+    prng_key=None,
+    debugger=None,
+    interface=None,
+) -> Result:
+    """Simulate a single shot of a single quantum script with native mid-circuit measurements.
+
+    Args:
+        circuit (QuantumTape): The single circuit to simulate
+        rng (Union[None, int, array_like[int], SeedSequence, BitGenerator, Generator]): A
+            seed-like parameter matching that of ``seed`` for ``numpy.random.default_rng``.
+            If no value is provided, a default RNG will be used.
+        prng_key (Optional[jax.random.PRNGKey]): An optional ``jax.random.PRNGKey``. This is
+            the key to the JAX pseudo random number generator. If None, a random key will be
+            generated. Only for simulation using JAX.
+        debugger (_Debugger): The debugger to use
+        interface (str): The machine learning interface to create the initial state with
+
+    Returns:
+        tuple(TensorLike): The results of the simulation
+        dict: The mid-circuit measurement results of the simulation
+    """
+    mcm_dict = {}
+    state, is_state_batched = get_final_state(
+        circuit, debugger=debugger, interface=interface, mid_measurements=mcm_dict
+    )
+    return (
+        measure_final_state(circuit, state, is_state_batched, rng=rng, prng_key=prng_key),
+        mcm_dict,
+    )
+
+
+def accumulate_native_mcm(circuit: qml.tape.QuantumScript, all_shot_meas, one_shot_meas):
+    """Incorporates new measurements in current measurement sequence.
+
+    Args:
+        circuit (QuantumTape): A one-shot (auxiliary) QuantumScript
+        all_shot_meas (Sequence[Any]): List of accumulated measurement results
+        one_shot_meas (Sequence[Any]): List of measurement results
+
+    Returns:
+        tuple(TensorLike): The results of the simulation
+    """
+    if not isinstance(one_shot_meas, Sequence):
+        one_shot_meas = [one_shot_meas]
```

Comment:
Is this the case for when a single measurement was squeezed?

```suggestion
    if len(circuit.measurements) == 1:
        one_shot_meas = [one_shot_meas]
```

Reply:
Yes.